transmogrifier/sources/json/aardvark.py

```diff
@@ -349,17 +349,41 @@ def get_publication_information(source_record: dict) -> list[str]:
         return publication_information
 
     @staticmethod
-    def get_rights(source_record: dict) -> list[timdex.Rights]:
+    def get_rights(source: str, source_record: dict) -> list[timdex.Rights]:
         """Get values from source record for TIMDEX rights field."""
         rights = []
 
         if "dct_accessRights_s" in source_record:
+            kind_access_rights = "Access"
+            kind_access_to_files = "Access to files"
             rights.append(
                 timdex.Rights(
-                    description=source_record["dct_accessRights_s"], kind="Access"
+                    description=source_record["dct_accessRights_s"],
+                    kind=kind_access_rights,
                 )
             )
 
+            if source == "gisogm":
+                rights.append(
+                    timdex.Rights(
+                        description="Not owned by MIT", kind=kind_access_to_files
+                    )
+                )
+            elif source == "gismit":
+                if source_record["dct_accessRights_s"] == "Restricted":
+                    rights.append(
+                        timdex.Rights(
+                            description="Requires MIT authentication",
+                            kind=kind_access_to_files,
+                        )
+                    )
+                else:
+                    rights.append(
+                        timdex.Rights(
+                            description="Free/open-access", kind=kind_access_to_files
+                        )
+                    )
```

Comment:
A couple observations:

1. First, I think the logic based on source + `dct_accessRights_s` looks good!  Removed from other comments, this was my understanding of how those values are determined.

2. Given that `dct_accessRights_s` is a **required** field for MITAardvark ([link to JSONSchema in GeoHarvester](https://github.com/MITLibraries/geo-harvester/blob/main/harvester/records/schemas/mit-schema-aardvark.json#L135)), this feels safe to have this nested under line #356 `if "dct_accessRights_s" in source_record` as we know they will always have it.  

My concern -- and it's not specific to this PR -- is how does someone looking at this code alone know that?  Curious @ehanson8's thoughts here, but what if this `get_rights()` method threw an exception if `dct_accessRights_s` is not present?  Then we could dedent everything under line #356 as we'd know it's present.  But, perhaps this is a can of worms as it would suggest similar treatment for other required fields.  And/or, make it brittle in the event we change what's required.  

In summary to #2: probably just fine as-is, but just noting what feels like an unresolved tension with checks for required MITAardvark fields.

3. If `gismit` and `gisogm` were broken out into their own transformations, we also avoid this `if/else` logic.  But again, not in scope for this PR, but just noting as we begin to get into more OGM record considerations.

Reply:
> My concern -- and it's not specific to this PR -- is how does someone looking at this code alone know that? Curious @ehanson8's thoughts here, but what if this `get_rights()` method threw an exception if `dct_accessRights_s` is not present? Then we could dedent everything under line #356 as we'd know it's present. But, perhaps this is a can of worms as it would suggest similar treatment for other required fields. And/or, make it brittle in the event we change what's required.

That feels like overkill if `geo-harvester` is doing any validation around this. We have to do that for outside sources we don't control (RDI sources e.g. WHOAS, Zenodo) but when we know `geo-harvester` considers it required, I don't think it's worth the added complexity of checking again

Reply:
> 3\. If `gismit` and `gisogm` were broken out into their own transformations, we also avoid this `if/else` logic.  But again, not in scope for this PR, but just noting as we begin to get into more OGM record considerations.

I think that is likely at this point gives the `get_source_link` logic but fine to leave the if/else here for now

Reply:
Hi @ghukill ! Thank you for your feedback and pointing out **2**. 

I did wonder about this and actually wanted to remove the condition `if "dct_accessRights_s" in source_record` (but forgot to do so). I did get a little confused as to why "rights" wasn't declared a required variable (if it's required in `MITAardvark`), _but_ remembered that TIMDEX doesn't only deal with geospatial metadata.

If we removed the conditional in `line 356`, I think the syntax of `source_record[<var>]` is sufficient in communicating to a programmer that "<var> is required in `source_record` because it is **indexing a value from the dictionary**". 🤔 

In terms of throwing an exception, I'm not sure if this makes sense, but does it make sense to...

I'll try to see what applying the suggestion in **3** looks like and we can see how it feels? 🤓

Reply:
Erg, my comment got deleted when I had to switch contexts.  Will try and keep it brief here.

I agree that explicit checking + exception throwing is overkill, for the reason that GeoHarvester + JSONSchema is ensuring that `dct_accessRights_s` MUST be present.  

To summarize all points, I'd propose:
1. remove `if "dct_accessRights_s" in source_record...` check
2. do not throw exception; handled by direct dictionary access which will throw an exception (but should never happen)

That addresses the majority of my observation(s) above.  What I think this achieves, is that then this newly added "Access to files" logic -- just by looking at code -- is gauranteed to be applied to every record for this `aardvark` transformation, which is what we need and want.

Reply:
💯 Totally agree with Graham's proposal